From 09b45c852dd17b381a3e0a4cef9ecc1ab813dc3f Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Wed, 15 Aug 2018 15:26:07 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399: don't enable hdmi_cec pin by
 default at soc dtsi

Fix pin conflicts when gpio4c7 is used for hdmi_cec and edp_hpd. CEC is an
optional function for HDMI. Thus the hdmi_cec pin should be specified
in the board .dts.

Without this patch we see below pin conflict when both hdmi and edp enabled.

[    0.969331] rockchip-pinctrl pinctrl: pin gpio4-23 already requested by
ff940000.hdmi; cannot claim for ff970000.edp
[    0.969350] rockchip-pinctrl pinctrl: pin-151 (ff970000.edp) status -22
[    0.969361] rockchip-pinctrl pinctrl: could not request pin 151 (gpio4-23)
from group edp-hpd on device rockchip-pinctrl
[    0.969371] rockchip-dp ff970000.edp: Error applying setting,
reverse things back

Change-Id: Id3e9d7fadb6a4cfd65827be9c4b55336406995f7
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 36fbafe89e19..0e3d7c9da3ee 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -1980,7 +1980,7 @@
 		compatible = "rockchip,rk3399-dw-hdmi";
 		reg = <0x0 0xff940000 0x0 0x20000>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&hdmi_i2c_xfer>, <&hdmi_cec>;
+		pinctrl-0 = <&hdmi_i2c_xfer>;
 		interrupts = <GIC_SPI 23 IRQ_TYPE_LEVEL_HIGH 0>;
 		clocks = <&cru PCLK_HDMI_CTRL>,
 			 <&cru SCLK_HDMI_SFR>,
-- 
2.35.3

